src/db/overrides.rs

```diff
@@ -0,0 +1,129 @@
+use crate::error::Result;
+use postgres::Client;
+use std::time::Duration;
+
+#[derive(Default, Debug, Clone, Copy, Eq, PartialEq)]
+pub struct Overrides {
+    pub memory: Option<usize>,
+    pub targets: Option<usize>,
+    pub timeout: Option<Duration>,
+}
+
+impl Overrides {
+    pub fn all(conn: &mut Client) -> Result<Vec<(String, Self)>> {
+        Ok(conn
+            .query("SELECT * FROM sandbox_overrides", &[])?
+            .into_iter()
+            .map(|row| (row.get("crate_name"), Self::from_row(row)))
+            .collect())
+    }
+
+    pub fn for_crate(conn: &mut Client, krate: &str) -> Result<Option<Self>> {
+        Ok(conn
+            .query_opt(
+                "SELECT * FROM sandbox_overrides WHERE crate_name = $1",
+                &[&krate],
+            )?
+            .map(Self::from_row))
+    }
+
+    fn from_row(row: postgres::Row) -> Self {
+        Self {
+            memory: row
+                .get::<_, Option<i64>>("max_memory_bytes")
+                .map(|i| i as usize),
+            targets: row.get::<_, Option<i32>>("max_targets").map(|i| i as usize),
+            timeout: row
+                .get::<_, Option<i32>>("timeout_seconds")
+                .map(|i| Duration::from_secs(i as u64)),
+        }
+    }
+
+    pub fn save(conn: &mut Client, krate: &str, overrides: Self) -> Result<()> {
```

Comment:
coming from my wrongly queued crates: 

should this validate if a crate with the given name exists?

Reply:
Makes sense, we're unlikely to be adding overrides before a crate's first failed build. Added it as a warning.